goodtables/cells.py

```diff
@@ -0,0 +1,73 @@
+from six.moves import zip_longest
+
+
+def create_cells(headers, schema_fields, values=None, row_number=None):
+    '''Create list of cells from headers, fields and values.
+
+    Args:
+        headers (List[str]): The headers values.
+        schema_fields (List[tableschema.field.Field]): The tableschema
+            fields.
+        values (List[Any], optional): The cells values. If not specified,
+            the created cells will have the same values as their
+            corresponding headers. This is useful for specifying headers
+            cells.
+            If the list has any `None` values, as is the case on empty
+            cells, the resulting Cell will have an empty string value. If
+            the `values` list has a different length than the `headers`,
+            the resulting Cell will have value `None`.
+        row_number (int): The row number.
+
+    Returns:
+        List[dict]: List of cells.
+    '''
+    fillvalue = '_fillvalue'
+    is_header_row = (values is None)
+    cells = []
+
+    iterator = zip_longest(headers, schema_fields, values or [], fillvalue=fillvalue)
+    for column_number, (header, field, value) in enumerate(iterator, start=1):
+        if header == fillvalue:
+            header = None
+        elif is_header_row:
+            value = header
+        if field == fillvalue:
+            field = None
+        if value == fillvalue:
+            value = None
+        elif value is None:
+            value = ''
+
+        cell = create_cell(header, value, field, column_number, row_number)
+        cells.append(cell)
+
+    return cells
+
+
+def create_cell(header, value=None, field=None, column_number=None, row_number=None):
+    cell = {
+        'header': header,
+        'field': field,
+        'value': value,
+        'column-number': column_number,
+        'number': column_number,  # FIXME: Remove deprecated "number"
+        'row-number': row_number,
+    }
+    return _remove_none_values(cell)
+
+
+def is_complete(cell):
+    return None not in [
+        cell.get('header'),
+        cell.get('field'),
+        cell.get('value'),
```

Comment:
That's not how the iteration protocol works - please read https://github.com/frictionlessdata/goodtables-py/issues/247

Reply:
You mean that this doesn't handle the case where `cell['header']` exists but is `None`? I can change this to `'header' in cell`, but then the following cell will be considered "complete":

```python
cell = {
  'header': None,
  'field': None,
  'value': None,
  'column-number': None
}
```

Which seems wrong, but I might be missing something.